psycopg3/types/datetime.py

```diff
@@ -0,0 +1,146 @@
+"""
+Adapters for datetime types.
+https://www.postgresql.org/docs/current/datatype-datetime.html#DATATYPE-DATETIME-TABLE
+"""
+
+# Copyright (C) 2020 The Psycopg Team
+
+import codecs
+import datetime
+import struct
+from typing import Tuple
+
+from ..adapt import Dumper, Loader
+from .oids import builtins
+
+_encode = codecs.lookup("ascii").encode
+_decode = codecs.lookup("ascii").decode
+
+
+@Dumper.text(datetime.date)
+@Dumper.binary(datetime.date)
+def dump_date(obj: datetime.date) -> Tuple[bytes, int]:
+    # TODO https://stackoverflow.com/questions/13468126/a-faster-strptime
+    # Optimize inspiration from isoformat
+    # https://github.com/python/cpython/blob/master/Lib/datetime.py#L849
+    # String Slicing
+    return _encode(str(obj))[0], builtins["date"].oid
+
+
+@Dumper.text(datetime.time)
+@Dumper.binary(datetime.time)
+def dump_time(obj: datetime.time) -> Tuple[bytes, int]:
+    if obj.tzinfo is None:
+        return _encode(str(obj))[0], builtins["time"].oid
+    else:
+        return _encode(str(obj))[0], builtins["timetz"].oid
+
+
+@Dumper.text(datetime.datetime)
+@Dumper.binary(datetime.datetime)
+def dump_datetime(obj: datetime.datetime) -> Tuple[bytes, int]:
+    if obj.tzinfo is None:
+        return _encode(str(obj))[0], builtins["timestamp"].oid
+    else:
+        return _encode(str(obj))[0], builtins["timestamptz"].oid
+
+
+@Dumper.text(datetime.timedelta)
+@Dumper.binary(datetime.timedelta)
+def dump_timedelta(obj: datetime.timedelta) -> Tuple[bytes, int]:
+    return _encode(str(obj))[0], builtins["interval"].oid
+
+
+@Loader.text(builtins["date"].oid)
+def load_date(data: bytes) -> datetime.date:
+    # TODO Multiple formats perhaps could do a try-except for loop?
+    return datetime.datetime.strptime(_decode(data)[0], "%Y-%m-%d").date()
+
+
+@Loader.binary(builtins["date"].oid)
+def load_date_binary(data: bytes) -> datetime.date:
+    dateADT = struct.unpack("!i", data)[0]  # num days since 2020-01-01
+    return datetime.date(year=2000, month=1, day=1) + datetime.timedelta(
```

Comment:
You can have `datetime.date(year=2000, month=1, day=1)` as a module constant and avoid this function call at every decode.